docker/Dockerfile-run

```diff
@@ -0,0 +1,51 @@
+####################################################################
```

Comment:
i tried to comment this pretty well just have a read over it.  the main idea is we use the multistage build to save space. i keep the debug info around but compressed in a targz. this brings the image down in size by about 1.5gb to a total of 1.1gb. downstream consumers of this image can always remove this extra debug info to make their image even slimmer.

also note that there is no `CMD` in the final image. the idea here is that this is the base for people to use to do whatever use-case they want to use valhalla for (building tiles, running a service, doing analytics, who knows).